pkg/controller/assignmetadata/assignmetadata_controller.go

```diff
@@ -142,10 +155,21 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	mutator, err := mutation.MutatorForAssignMetadata(assignMetadata)
 	if err != nil {
 		log.Error(err, "Creating mutator for resource failed", "resource", request.NamespacedName)
+		tracker.CancelExpect(assignMetadata)
 		return ctrl.Result{}, err // TODO: reque all request on error now. change this once a decision is made on how to process errors
 	}
-	if err := r.system.Upsert(mutator); err != nil {
-		log.Error(err, "Insert failed", "resource", request.NamespacedName)
+	if !deleted {
+		if err := r.system.Upsert(mutator); err != nil {
+			log.Error(err, "Insert failed", "resource", request.NamespacedName)
+			tracker.CancelExpect(assignMetadata)
```

Comment:
Now that #1014 is merged, this should be `TryCancelExpect`

Reply:
Done